lib/rules/no-callback-leaks-in-ember-objects.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+// const ember = require('../utils/ember');
+const utils = require('../utils/utils');
+
+
+//------------------------------------------------------------------------------
+// Ember object rule - Avoid callback memory leaks
```

Comment:
Should this rule be limited to Ember objects only?

If it should, it would need a `if (!(ember.isEmberObject(node) || ember.isReopenObject(node)))` check somewhere, similar to the existing `avoid-leaking-state-in-ember-objects` rule.

If it's not necessary to limit it to Ember objects, and we can just run it on a per-file basis, then we would want to rename the rule to avoid mentioning `in-ember-objects`, and remove all the extra Ember object code from the test cases.

My preference is to avoid limiting it to Ember objects if we don't need to.